dts/bindings/flash_controller/renesas,smartbond-flash-controller.yaml

```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2022 Renesas Electronics Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+description: Renesas SmartBond(tm) family flash controller
+
+compatible: "renesas,smartbond-flash-controller"
+
+include: flash-controller.yaml
+
+properties:
+  qspif_base_address:
+    type: int
+    required: true
+    description: Base address of QSPIF in memory map
```

Comment:
This looks strange. Why not just take a phandle to the QSPIF so you can use DT_REG_ADDR on that node?

Reply:
The "flash" node in device tree is located at 0x0 because this is where internal bootloader remaps it after boot so interrupt vector is at 0x0. Flash controller has to access memory at 0x16000000 instead of 0x0 so this is the offset (i.e. qspif-base-address) I need here. If I set "flash" node at 0x16000000, the VTOR will be also set to 0x16000000 which is ro, so any attempt to modify interrupt vector will result in a fault.

So not sure what you mean by taking a handle to QSPIF - should I add a separate QSPIF node to device tree?

Reply:
QSPIF can be described in Devicetree, so you can take its address from its node.